docs/education/dlt-fundamentals-course/lesson_2_dlt_sources_and_resources_create_first_dlt_pipeline.py

```diff
@@ -0,0 +1,719 @@
+
+    In this exercise, you'll build a dlt pipeline to fetch data from the GitHub REST API. The goal is to learn how to use `dlt.pipeline`, `dlt.resource`, and `dlt.source` to extract and load data into a destination.
+
+    ## Instructions
+
+    1. **Explore the GitHub API**
+
+       Visit the [GitHub REST API Docs](https://docs.github.com/en/rest) to understand the endpoint to [list public repositories](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28) for an organization:
+
+       `GET https://api.github.com/orgs/{org}/repos`
+
+    2. **Build the pipeline**
+
+       Write a script to:
+
+       - Fetch repositories for the **dlt-hub** organization.
+       - Use `dlt.resource` to define the data extraction logic.
+       - Combine all resources into a single `@dlt.source`.
+       - Load the data into a DuckDB database.
+
+    3. **Inspect the data**
+
+       Use a `duckdb` connection, `sql_client`, or `pipeline.dataset()`.
+
+    > **Note**: For this exercise you don't need to use authentication or pagination.
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Play with the API using the requests library:""")
+    return
+
+
+@app.cell
+def _(requests):
+    response = requests.get('https://api.github.com/orgs/dlt-hub/repos')
+    response.json()[0]
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""In the code snippet below, you will find an **example** for the **`events`** endpoint:"""
+    )
+    return
+
+
+@app.cell
+def _(DltResource, Iterable, TDataItems, dlt, requests):
+    @dlt.resource
+    def github_events() -> TDataItems:
+        url = 'https://api.github.com/orgs/dlt-hub/events'
+        response = requests.get(url)
+        yield response.json()
+
+    @dlt.source
+    def github_data() -> Iterable[DltResource]:
+        return (github_events,)
+    github_pipeline = dlt.pipeline(pipeline_name='github_pipeline', destination='duckdb', dataset_name='github_data')
+    _load_info = github_pipeline.run(github_data())
+    print(_load_info)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    ### Question
+    How many columns has the `github_repos` table? Use a `duckdb` connection, `sql_client` or `pipeline.dataset()`.
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    ## **Exercise 2: Create a pipeline for the GitHub API – stargazers endpoint**
+
+    Create a `dlt.transformer` for the **"stargazers"** endpoint
+    `https://api.github.com/repos/OWNER/REPO/stargazers` for the `dlt-hub` organization.
+
+    Use the `github_repos` resource as the main resource for the transformer:
+
+    1. Get all repositories in the `dlt-hub` organization.
+    2. Feed these repository names into the `dlt` transformer and retrieve all stargazers for all `dlt-hub` repositories.
+    """)
+    return
+
+
+@app.cell
+def _():
+    print("HERE IS YOUR CODE")
```

Comment:
Maybe brush up the placeholder text
```suggestion
    print("YOUR CODE GOES HERE")
```